Source/FieldSolver/FiniteDifferenceSolver/MacroscopicEvolveHM_2nd.cpp

```diff
@@ -965,8 +1016,17 @@ void FiniteDifferenceSolver::MacroscopicEvolveHMCartesian_2nd(
 
                         [=] AMREX_GPU_DEVICE(int i, int j, int k) {
 
-                            Real mag_Ms_arrz = MacroscopicProperties::macro_avg_to_face(i,j,k,Mzface_stag,mag_Ms_arr);
-
+                            amrex::Real x, y, z;
+                            x = problo[0] + i*dx[0] + (1._rt - Mzface_stag[0]) * dx[0]*0.5;
```

Comment:
can you call GetCellCoordinates() here too. so we reduce code duplication

Reply:
Yes, addressed this.